osversion/platform_compat_windows.go

```diff
@@ -0,0 +1,71 @@
+package osversion
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// List of stable ABI compliant
+// ltsc releases
+// Note: List must be sorted
+var compatLTSCReleases = []uint{
+	V21H2Server,
+}
+
+// Checks if given host and guest OS versions are compatible.
+// It includes support for stable ABI compliant OS versions.
+// Every release after WS 2022 will support the previous
+// ltsc container image. Stable ABI is in preview mode
+// for windows 11 client.
+// Refer: https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2022%2Cwindows-10#windows-server-host-os-compatibility
+func AreHostAndGuestCompatible(hostOSVersion, guestOSVersion string) bool {
```

Comment:
I actually think this function should not take a string of dot-separated numbers. This forces the caller to produce a version string, even if they are internally working with raw numbers. I would suggest one of the following:
- Scope this function to _just_ checking the build number, and only take build number as an input. It would fall on the caller to validate major/minor versions.
- Pass some sort of struct (like `OSVersion`) that has numbers inside of it.

Both of these have the advantage that we no longer have to do any string parsing in here.